test/testdata/deployednettemplates/generate-recipe/generate_network_tpl.py

```diff
@@ -0,0 +1,204 @@
+
+v2 format:
+
+    ```
+    group1 group2 minrtt
+    ```
+
+    > group1 and group2 are referring to individual hosts (e.g. R1 and R2), so there should be one group per host.
+
+"""
+import argparse
+import json
+import math
+
+DEFAULT_NUM_N = 5
+DEFAULT_NUM_NPN = 5
+DEFAULT_REGION = 'us-west-1'
+
+def main():
+    args = parse_args()
+
+    # initialize network_tpl with defaults
+    network_tpl = get_default_network_tpl()
+
+    groups = []
+
+    if args.network_rules_file is not None:
+        network_tpl_from_rules = gen_network_tpl_from_rules_v2(args.network_rules_file)
+        merge(network_tpl_from_rules, network_tpl)
+
+    # write network_tpl to file
+    with open(args.out, 'w') as out:
+        out.write(json.dumps(network_tpl, indent=2))
+        out.write('\n')
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Generate a network-tpl.json file for generate_network.py'
+    )
+    parser.add_argument('-n', '--network-rules-file', help='Path of network_performance_rules file', required=True)
+    parser.add_argument('-o', '--out', help='Path to write output', default='network-tpl.json')
+    return parser.parse_args()
+
+
+def get_default_network_tpl():
+    return {
+        'network': {
+            'wallets': 5
+        },
+        'instances': {
+            'relays': {
+                'config': './configs/relay.json',
+                'type': 'c5.xlarge',
+                'count': 5
+            },
+            'participatingNodes': {
+                'config': './configs/node.json',
+                'type': 'c5.xlarge',
+                'count': 5
+            },
+            'nonParticipatingNodes': {
+                'config': './configs/nonPartNode.json',
+                'type': 'c5.xlarge',
+                'count': 5
+            }
+        }
+    }
+
+def merge(source, destination):
+    for key, value in source.items():
+        if isinstance(value, dict):
+            # get node or create one
+            node = destination.setdefault(key, {})
+            merge(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+def gen_network_tpl_from_rules_v2(path):
+    """
+        Loads network_performance_rules v2 file, and generates a network-tpl.json file
+        @param path: the filesystem path to the network_performance_rules file
+        @return list of network-tpl.json groups
+    """
+    groups = []
+
+    with open(path) as network_performance_rules:
+        npr = network_performance_rules.readlines()
+
+    found = []
+    num_relays = 0
+    num_npn = 0
+    num_n = 0
+
+    # loop over rules and get counts of types of instances
+    for rule in npr:
+        # algonet retrieves groups from terraform-inventory and they are lowercase
+        name = rule.split(' ')[0].lower()
+        relays = 0
+        nonParticipatingNodes = 0
+        participatingNodes = 0
+
+        if name in found:
```

Comment:
make `found` to be `set()` => const time lookup vs linear, could be noticeable with 20+k rules file lines